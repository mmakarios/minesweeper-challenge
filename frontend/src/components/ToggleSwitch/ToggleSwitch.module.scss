@import "../../styles/variables";

.toggleSwitch {
  display: flex;
  align-items: center;
  > input {
    height: 0;
    width: 0;
    visibility: hidden;
    &:checked + label:after {
      left: calc(100% - 5px);
      transform: translateX(-100%);
    }
  }

  > label {
    cursor: pointer;
    width: 5rem;
    height: 2.5rem;
    background: var(--toggle-switch-background);
    display: block;
    border-radius: 100px;
    position: relative;
    &:after {
      content: "";
      position: absolute;
      top: 5px;
      left: 5px;
      width: 1.875rem;
      height: 1.875rem;
      background: var(--toggle-switch-circle);
      border-radius: 100%;
      transition: 0.3s;
    }
    &:active:after {
      width: 3.125rem;
    }
  }
}
