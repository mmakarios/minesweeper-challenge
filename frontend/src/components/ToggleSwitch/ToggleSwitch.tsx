import styles from "./ToggleSwitch.module.scss";

type Props = React.HTMLProps<HTMLInputElement> &
  React.InputHTMLAttributes<HTMLInputElement> & {
    onClick(): void;
  };

export const ToggleSwitch = ({ onClick, ...rest }: Props) => {
  return (
    <div className={styles.toggleSwitch}>
      <input type="checkbox" {...rest} />
      <label onClick={onClick} />
    </div>
  );
};

export default ToggleSwitch;
