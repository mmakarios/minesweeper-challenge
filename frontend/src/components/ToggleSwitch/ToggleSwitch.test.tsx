import React from "react";
import renderer from "react-test-renderer";
import ToggleSwitch from "./ToggleSwitch";

it("renders component", () => {
  const tree = renderer.create(<ToggleSwitch onClick={() => {}} />).toJSON();

  expect(tree).toMatchSnapshot();
});
