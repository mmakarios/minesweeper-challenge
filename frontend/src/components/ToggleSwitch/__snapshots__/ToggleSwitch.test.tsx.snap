// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div
  className="toggleSwitch"
>
  <input
    type="checkbox"
  />
  <label
    onClick={[Function]}
  />
</div>
`;
